articles/decoding-jwt-structure.md

```diff
@@ -0,0 +1,276 @@
+# Decoding JWTs Structure
+
+By Rishi Raj Jain
+
+<insert-fig>
+
+JSON Web Tokens (JWTs) are compact, URL-safe tokens commonly used in modern authentication and authorization systems, particularly with OAuth2 flows. This guide provides a technical deep-dive into JWT architecture, examining each component and its role in the token's security and functionality.
+
+JWTs are sometimes incorrectly referred to as "JWT Tokens" - this would expand to "JSON Web Token Token," which is redundant. We'll use "JWT" throughout this guide. Additionally, while JWTs are frequently used for authentication, they're technically just base64-encoded JSON containers that can serve various purposes beyond auth. This guide focuses on their authentication use case while explaining the underlying structure.
+
+Let's examine a real JWT example. Here's a freshly generated token with line breaks added for readability (normally JWTs are single-line strings):
+
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImY1ODg5MGQxOSJ9.eyJhdWQiO
```

Comment:
make a single line with no line breaks to make copying easier

Reply:
Token should be updatd to reflect this data: 

{
  "aud": "85a03867-dccf-4882-adde-1a79aeec50df",
  "exp": 1644884185,
  "iat": 1644880585,
  "iss": "acme.com",
  "sub": "00000000-0000-0000-0000-000000000001",
  "jti": "3dd6434d-79a9-4d15-98b5-7b51dbb2cd31",
  "authenticationType": "PASSWORD",
  "email": "admin@phasetwo.io",
  "email_verified": true,
  "applicationId": "85a03867-dccf-4882-adde-1a79aeec50df",
  "roles": ["ceo"]
}